core/src/main/java/org/opensearch/sql/calcite/CalciteRelNodeVisitor.java

```diff
@@ -997,13 +997,54 @@ private Pair<List<RexNode>, List<AggCall>> aggregateWithTrimming(
     // because that Mapping only works for RexNode, but we need both AggCall and RexNode list.
     Pair<List<RexNode>, List<AggCall>> reResolved =
         resolveAttributesForAggregation(groupExprList, aggExprList, context);
-
+    List<String> names = getGroupKeyNamesAfterAggregation(reResolved.getLeft());
     context.relBuilder.aggregate(
         context.relBuilder.groupKey(reResolved.getLeft()), reResolved.getRight());
-
+    // During aggregation, Calcite projects both input dependencies and output group-by fields.
+    // When names conflict, Calcite adds numeric suffixes (e.g., "value0").
+    // Apply explicit renaming to restore the intended aliases.
+    if (names.size() == reResolved.getLeft().size()) {
```

Comment:
when the `names.size` not equals to `reResolved.getLeft().size()`? seems the condition is always true